src/main/java/graphql/execution/directives/QueryDirectives.java

```diff
@@ -32,6 +33,13 @@
      */
     Map<String, List<GraphQLDirective>> getImmediateDirectivesByName();
 
+    /**
+     * This will return a map of the applied directives that are immediately on a merged field
+     *
+     * @return a map of all the applied directives immediately on this merged field
+     */
+    Map<String, List<GraphQLAppliedDirective>> getImmediateAppliedDirectivesByName();
```

Comment:
so here the order of the keys matter, right? We should communicate this clearly that the `Map` keys order reflects the order of the directives found. 

Mh ... actually is that true with a merged field?

Reply:
`Field -> AppliedDirectives` is super weird in another way.   That is we are kinda saying - we can't decide so.... here you go.

I mean we are more correct - only because we are no overly useful.

Reply:
The code is

```
            final Map<Field, List<GraphQLDirective>> byField = new LinkedHashMap<>();
            final Map<Field, List<GraphQLAppliedDirective>> byFieldApplied = new LinkedHashMap<>();
            mergedField.getFields().forEach(field -> {
                List<Directive> directives = field.getDirectives();
                ImmutableList<GraphQLDirective> resolvedDirectives = ImmutableList.copyOf(
                        directivesResolver
                                .resolveDirectives(directives, schema, variables)
                                .values()
                );
                byField.put(field, resolvedDirectives);
                // at some point we will only use applied
                byFieldApplied.put(field, ImmutableKit.map(resolvedDirectives, GraphQLDirective::toAppliedDirective));
            });
```

So its in Merged field order - but that in itself might not be meaningful.

I guess this is NO worse than before in directive terms - so we can improve it later with a `Field -> AppliedDirectives` if we choose